### PR TITLE
Add hastebin as implementation of gist interface

### PR DESCRIPTION
**Pull request by drvan**
Hastebin is an adequate alternative to github anonymous gists, which were deprecated recently. This PR adds hastebin as an implementation of the gist interface, and modifies crbot to use remotehastebin instead of remotegist.

### CODE REVIEW

File: crbot.go [hunk 1]
```diff
@@ -48,7 +48,7 @@ func main() {
 	karmaMap := stringmap.NewRedisStringMap(redisClient, RedisKarmaHash)
 	voteMap := stringmap.NewRedisStringMap(redisClient, RedisVoteHash)
 
-	gist := api.NewRemoteGist()
```
Comment (jakevoytko) [inline crbot.go:51]: Delete NewRemoteGist and the RemoteGist implementation

File: api/remotehastebin.go [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+	defer response.Body.Close()
+
+	responseMap := map[string]interface{}{}
+	if err := json.NewDecoder(response.Body).Decode(&responseMap); err != nil {
+		log.Info("Error reading Hastebin response", err)
+		return "", errors.New(MsgHastebinResponseFail)
+	}
+
+	if finalUrl, ok := responseMap["key"]; ok {
+		return "https://hastebin.com/" + finalUrl.(string), nil
```
Comment (jakevoytko) [inline api/remotehastebin.go]: Let's link to `/raw/`... the highlighted version highlights a bunch of weird stuff, and this looks a lot better

https://hastebin.com/raw/wemopamusa

File: api/remotehastebin.go [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+	if err != nil {
+		log.Info("Error POSTing Hastebin", err)
+		return "", errors.New(MsgHastebinPostFail)
+	} else if response.StatusCode != 200 {
+		log.Info("Bad status code", errors.New("Code: "+strconv.Itoa(response.StatusCode)))
+		body, _ := ioutil.ReadAll(response.Body)
+		log.Info("Response body: ", errors.New(string(body)))
+		return "", errors.New(MsgHastebinStatusCode)
+	}
+	defer response.Body.Close()
```
Comment (jakevoytko) [inline api/remotehastebin.go]: Oo, good catch. In the documentation, it looks like you need to call this after checking the error, so mind moving this after the error check and before the status code check?

**Review by jakevoytko — COMMENTED** [review]
Just want a few small changes

 - Error checking cleanup 
 - link to `/raw/`
 - Delete the old implementation

I tested this and it worked, so I'll be able to push this when the cleanups are in

Thanks!!

**Comment by drvan** [comment]
Made the requested changes - let me know if anything else is needed.

**Pull request merged**